### ISSUE TITLE
Segmentation fault in longjmp

### ISSUE THREAD

**Issue by renatahodovan** [Samsung/jerryscript#341]
###### Jerry version:
    Checked revision: 146ac15
    Build: debug.linux
###### OS: 
    Ubuntu 15.04, x86_64
###### Test case:
```javascript
�
```

Hex: BF 0A 0A

###### Backtrace:
```javascript

^
ERROR: Ln 1, Col 1: Invalid source encoding

Program received signal SIGSEGV, Segmentation fault.
0x0000000000497a84 in longjmp () at jerryscript/jerry-libc/target/linux/jerry-asm.S:69
69	  _LONGJMP
(gdb) bt
# 0  0x0000000000497a84 in longjmp () at jerryscript/jerry-libc/target/linux/jerry-asm.S:69
Backtrace stopped: Cannot access memory at address 0x0

```

**Comment by ruben-ayrapetyan** [comment]
The failure is caused by invocation of `syntax_raise_error` before `setjmp` was performed in `parser_parse_program`.

**Comment by ruben-ayrapetyan** [comment]
Pull request with fix: #390

**Comment by ruben-ayrapetyan** [comment]
Fixed.
